plugins/modules/ome_network_vlan_info.py

```diff
@@ -0,0 +1,250 @@
+            "Type": {
+                "Description": "This is the network for general purpose traffic. QOS Priority : Silver.",
+                "Id": 2,
+                "Name": "General Purpose (Silver)",
+                "NetworkTrafficType": "Ethernet",
+                "QosType": {
+                    "Id": 3,
+                    "Name": "Silver"
+                },
+                "VendorCode": "GeneralPurpose"
+            },
+            "UpdatedBy": null,
+            "UpdatedTime": "2020-09-02 18:49:11.507",
+            "VlanMaximum": 112,
+            "VlanMinimum": 112
+        }
+    ]
+}
+error_info:
+  description: Details of the HTTP Error.
+  returned: on HTTP error
+  type: dict
+  sample: {
+    "error": {
+      "code": "Base.1.0.GeneralError",
+      "message": "A general error has occurred. See ExtendedInfo for more information.",
+      "@Message.ExtendedInfo": [
+        {
+          "MessageId": "GEN1234",
+          "RelatedProperties": [],
+          "Message": "Unable to process the request because an error occurred.",
+          "MessageArgs": [],
+          "Severity": "Critical",
+          "Resolution": "Retry the operation. If the issue persists, contact your system administrator."
+        }
+      ]
+    }
+  }
+'''
+
+import json
+from ssl import SSLError
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.dellemc.openmanage.plugins.module_utils.ome import RestOME
+from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
+from ansible.module_utils.urls import ConnectionError, SSLValidationError
+
+# Base URI to fetch all logical networks information
+NETWORK_VLAN_BASE_URI = "NetworkConfigurationService/Networks"
+NETWORK_TYPE_BASE_URI = "NetworkConfigurationService/NetworkTypes"
+QOS_TYPE_BASE_URI = "NetworkConfigurationService/QosTypes"
+
+# Module Success Message
+MODULE_SUCCESS_MESSAGE = "Successfully retrieved the network VLAN information."
+
+# Module Failure Messages
+MODULE_FAILURE_MESSAGE = "Failed to retrieve the network VLAN information."
+NETWORK_VLAN_NAME_NOT_FOUND = "Provided network VLAN with name - '{0}' does not exist."
+
+SAFE_MAX_LIMIT = 9999
+
+
+def clean_data(data):
+    """
+    data: A dictionary.
+    return: A data dictionary after removing items that are not required for end user.
+    """
+    for k in ['@odata.id', '@odata.type', '@odata.context', '@odata.count']:
+        data.pop(k, None)
+    return data
+
+
+def get_network_type_and_qos_type_information(rest_obj, network_vlan):
+    """
+    rest_obj: Object containing information about connection to device.
+    network_vlan: A dictionary containing information of network VLAN.
+    return: updated dictionary with additional info for "Type" and "QosType" keys.
+    """
+    network_vlan = clean_data(network_vlan)
+    network_types_uri = "{0}({1})".format(NETWORK_TYPE_BASE_URI, network_vlan.get("Type"))
+    resp = rest_obj.invoke_request('GET', network_types_uri)
```

Comment:
@deepakjoshishri  Store the values of NetworkTypes and QosTypes using below URI once
"NetworkConfigurationService/NetworkTypes"
"NetworkConfigurationService/QosTypes"
and apply to the whole VLAN list which is fetched before? So that it will avoid 2 request call for each VLAN items.

Reply:
@Sajna-Shetty thanks for the comment, this helped me improve module performance by almost 1000x in the worst case when 4000+ VLAN(s) are present, further I have improved some performance by incorporating dictionary based approach to update each network VLAN info by network type and QoS type info.